app/controllers/host_aggregate_controller.rb

```diff
@@ -31,6 +31,36 @@ def new
     )
   end
 
+  def create
+    assert_privileges("host_aggregate_new")
+  case params[:button]
+  when "cancel"
```

Comment:
@MelsHyrule you need only assert privileges line, you don't need case loop here, because , you have already written add/save/cancel buttons code in react form.

Reply:
the definition would then be just 
``` 
 def create
    assert_privileges("host_aggregate_new")
  end
```

Reply:
you need @in_a_form = true line as well